### PR TITLE
Refactor SetMurderedByAsync into MurderPlayerAsync

### PR DESCRIPTION
**Pull request by js6pak**
### Description

Refactor MurderPlayer so it makes sense

### CODE REVIEW

File: src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs [hunk 1]
```diff
@@ -101,38 +101,30 @@ public async ValueTask SendChatToPlayerAsync(string text, IInnerPlayerControl? p
...
 
-            if (PlayerInfo.IsDead)
+            if (target.PlayerInfo.IsDead)
             {
-                return;
+                throw new ImpostorProtocolException("Tried to murder a player, but target was not alive.");
             }
 
-            // Update player.
             Die(DeathReason.Kill);
```
Comment (Mr-VIT) [inline src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs:121]:
```suggestion
            target.Die(DeathReason.Kill);
```

Comment (js6pak) [reply src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs:121]: whoops

File: src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs [hunk 2]
```diff
@@ -101,38 +101,30 @@ public async ValueTask SendChatToPlayerAsync(string text, IInnerPlayerControl? p
...
-            if (PlayerInfo.IsDead)
+            if (target.PlayerInfo.IsDead)
             {
-                return;
+                throw new ImpostorProtocolException("Tried to murder a player, but target was not alive.");
             }
 
-            // Update player.
-            Die(DeathReason.Kill);
+            ((InnerPlayerControl)target).Die(DeathReason.Kill);
```
Comment (js6pak) [inline src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs:121]: how bad is this?

**Comment by kantesilvestrs** [comment]
Can we just make sure that there is **'UGE** breaking changes notification added to the docs? I mean when this gets pushed - it will instantly break my plugin 😄 

Not to mention ==> people using docker will get the nightly update on next server restart and this will crash their servers, i.e. **me** <---

Do we really need to refactor existing function, can't we just add **deprecated** flag and add this method in parallel?

**Comment by js6pak** [comment]
Api is experimental so I don't we should care about that yet, and this method is just awful right now

**Pull request closed without merging**